composer/workflows/airflow_db_cleanup.py

```diff
@@ -429,17 +413,14 @@ def cleanup_function(**context):
             session.commit()
 
             list_dags = [str(list(dag)[0]) for dag in dags] + [None]
-            for dag in list_dags:
+            for dag_id in list_dags:
                 query = build_query(
-                    session,
-                    airflow_db_model,
-                    age_check_column,
-                    max_date,
-                    keep_last,
-                    keep_last_filters,
-                    keep_last_group_by,
+                    session=session,
+                    airflow_db_model=airflow_db_model,
+                    age_check_column=age_check_column,
+                    max_date=max_date,
+                    dag_id=dag_id,
                 )
-                query = query.filter(airflow_db_model.dag_id == dag)
```

Comment:
I am confused.
If we do not have this filter by dag_id here, then for all tables != DagRun we will not have filter on dag_id?
Should we have filter inside build_query method?

Reply:
Yes, my intention was to have this inside `build_query` block, but I missed that later.
This way it still works, but it makes sense to filter by dag. 
PTAL (when dag_id is present we filter by it).